targets/TARGET_TOSHIBA/TARGET_TMPM46B/trng_api.c

```diff
@@ -0,0 +1,105 @@
+/* mbed Microcontroller Library
+ * (C)Copyright TOSHIBA ELECTRONIC DEVICES & STORAGE CORPORATION 2017 All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "mbed_error.h"
+#include "tmpm46B_esg.h"
+#include "trng_api.h"
+
+#define ESG_BUSY_SET                  (0x00000001U)
+
+volatile uint32_t CompleteFlag = 0U; // Global variables
+
+static void ESG_Config(void)
+{
+    uint16_t latchtiming;
+    uint16_t Fintming = 560U;
+    // Confirm the ESG core stops
+    if (ESG_GetCalculationStatus() == ESG_CALCULATION_COMPLETE) {
+        // Confirm no interrupt generation
+        if (ESG_GetIntStatus() == DISABLE) {
+            // Set the latch timing
+            ESG_SetLatchTiming(ESG_LATCH_TIMING_1);
+            // Get latchtiming value
+            latchtiming = ESG_GetLatchTiming();
+            if (Fintming >= 512U * (latchtiming + 1U) + 3U) {
+                // Set the output timing
+                ESG_SetFintiming(Fintming);
+            } else {
+                error("Fintming value input error !\n");
+            }
+        } else {
+            // Do nothing
+        }
+    } else {
+        // Do nothing 
+    }
+}
```

Comment:
Unfortunately, the current API doesn't allow `trng_init` to have a return value to signal failure (that has been resolved in Mbed TLS 2.7.0 and will hopefully soon be merged into Mbed OS). Still, I'd like to seek for a solution which allows to distinguish initialization failure from success, in order to not have the TRNG driver potentially output non-random data because of unnoticed failure during initialization. 

Suggestion: Currently, the TRNG context `trng_s` consists of an unused `dummy` field only. This can be changed to a boolean flag indicating successful initialization, which is set on a successful call to `trng_init`, cleared on a call to `trng_free`, and checked for in `trng_get_bytes`.

Reply:
Thank you for the suggestion. we have added the errors handling during TRNG initialization to prevent accidental output of non-random data.